### PR TITLE
refactor(anolisa): inject systemd runner

### PR DESCRIPTION
**Pull request by ikunkun-sys**
## Why

Systemd operations currently spawn `systemctl` directly and collapse spawn failures,
non-zero exits, and unit state into string-based errors. This first R4 slice introduces
an injectable process boundary so system service behavior can be tested without mutating
the host while retaining structured failure evidence.

## What changed

- Add `Systemd<R: CommandRunner>` with the real runner used in production.
- Preserve operation, unit, exit code, stdout, and stderr in typed systemd failures.
- Migrate system-helper status and telemetry enable/disable callers to the boundary.
- Add fake-runner tests for status parsing, missing units, spawn failures, non-zero exits,
  and deferred disable behavior.

## Related issue

no-issue: bounded R4 refactoring slice with no user-facing feature request

## User / Agent impact

Normal enable, disable, and status command arguments remain unchanged. If `systemctl`
cannot be spawned, system-helper status now reports `unknown` instead of misclassifying
the host failure as a failed unit. Systemctl diagnostics are normalized with `LC_ALL=C`.

## Risk and compatibility

- [x] Public CLI, API, configuration, or documented behavior changed
- [ ] Privileged or security-sensitive behavior changed
- [ ] Cross-component contract changed
- [ ] Migration or rollback guidance is needed

Low risk. The systemctl argv and success-path parsing are unchanged. The source-level
API change is limited to the unpublished `anolisa-platform` workspace crate, and every
repository caller is migrated in this change. The checked item records the corrected
`failed` to `unknown` status classification for process-spawn failures.

## Validation

Validated on Linux from `src/anolisa`:

- `cargo fmt --all -- --check`
- `cargo clippy --workspace --all-targets --locked -- -D warnings`
- `cargo test --workspace --locked`
- `cargo doc --workspace --no-deps`

The workspace tests include ten isolated systemd boundary tests. No privileged host
service mutation was performed.

## Documentation and rollback

No documentation changes are included because this is an internal refactoring slice.
Rollback is a direct revert of this commit; no state or configuration migration is needed.

### CODE REVIEW

**Review by kongche-jbw — COMMENTED** [review]
Review baseline: `6c2e5eaa5c9972565032f121265a82582923d6c0...98ff0508d86d743886213798d1a466900d65da6b`

**[P2] Do not infer unit state from process diagnostics**

`src/anolisa/crates/anolisa-platform/src/systemd.rs:216-243` maps any non-zero
diagnostic containing `no such file or directory` to `NotFound`, and any remaining
diagnostic containing `failed` to `UnitFailed`. For example, `systemctl show` can return
`Failed to connect to bus: No such file or directory`; `check_service_state` then reports
`not installed`, although the unit was never queried. With `Host is down`, the same bus
failure becomes `failed`. This defeats the typed distinction and misreports container or
chroot hosts.

Possible direction: derive domain state from successful `LoadState`/`ActiveState`
properties and keep non-zero transport or process errors as `NonZeroExit`. Please add
regressions for both bus-error forms and a real `ActiveState=failed` result.

File: src/anolisa/crates/anolisa-platform/src/systemd.rs [hunk 1]
```diff
@@ -32,152 +66,414 @@ pub struct UnitStatus {
...
+    let failure = Box::new(SystemdCommandFailure {
+        operation: operation.to_string(),
+        unit: unit.map(str::to_string),
+        code: out.code,
+        stdout: out.stdout,
+        stderr: out.stderr,
+        detail,
+    });
+    if lower.contains("failed") {
+        return SystemdError::UnitFailed(failure);
```
[the remark this answers is not in the document]
Comment (ikunkun-sys) [reply src/anolisa/crates/anolisa-platform/src/systemd.rs]: Resolved in latest revision (`73628153`): `unit_status` now derives missing and failed states only from successful `LoadState`/`ActiveState` properties, while non-zero results remain `NonZeroExit`. Regressions cover both bus-error forms and `ActiveState=failed`; all ANOLISA gates pass.

**Review by kongche-jbw — COMMENTED** [review]
Review baseline: `265b133b0e54d786ef5b792828f8109d9ea751b9...73628153056a609c9c1f579ace3eaa84a74e4877`

未发现阻塞问题。

Remaining risks:

- 本地定向复跑因 Rust toolchain 组件恢复失败，未形成独立本地测试结果。

Verification scope:

- 检查了完整 4 文件 diff、相邻调用方与测试、已解决的上一轮 thread，以及当前 CI。
- CI `Test anolisa` 已通过格式、clippy、doc tests 和 workspace tests。

**Review by kongche-jbw — APPROVED** [review]
